library/src/main/java/com/novoda/downloadmanager/NotificationDispatcher.java

```diff
@@ -0,0 +1,45 @@
+package com.novoda.downloadmanager;
+
+import static com.novoda.downloadmanager.DownloadBatchStatus.Status.DELETION;
+import static com.novoda.downloadmanager.DownloadBatchStatus.Status.DOWNLOADED;
+
+class NotificationDispatcher {
+
+    private final Object waitForDownloadService;
+    private final NotificationCreator<DownloadBatchStatus> notificationCreator;
+
+    private DownloadService downloadService;
+
+    NotificationDispatcher(Object waitForDownloadService, NotificationCreator<DownloadBatchStatus> notificationCreator) {
+        this.waitForDownloadService = waitForDownloadService;
+        this.notificationCreator = notificationCreator;
+    }
+
+    void updateNotification(DownloadBatchStatus downloadBatchStatus) {
+        WaitForDownloadService.<Void>waitFor(downloadService, waitForDownloadService)
```

Comment:
Is this a blocking call?

Reply:
It's only blocking the current thread if the `download-service` doesn't exist. Given that the `Downloader` would also be blocked I don't think this is an issue.

Reply:
Ah right sorry. This is blocking yes. The `Downloader` waits for the service before performing the `download` on the `DownloadBatch` the batch then calls through to this dispatcher. So from what I understand this will all be on the `executor`. Correct me if I am wrong.

Reply:
Yes you are right, shall we use the working thread annotation? ;)

Reply:
0c35ee2